mlflow/llama_index/serialize_objects.py

```diff
@@ -148,17 +145,21 @@ def _deserialize_dict_of_objects(path: str) -> Dict[str, any]:
 
 def _serialize_dict_of_objects(dict_of_objects: Dict[str, object], path: str) -> None:
     to_serialize = {}
+    unsupported_objects = []
 
     for k, v in dict_of_objects.items():
         object_json = [object_to_dict(vv) for vv in v] if isinstance(v, list) else object_to_dict(v)
 
         if object_json == {}:
-            _logger.info(
-                f"{k} serialization is not supported. It will not be logged with your model"
-            )
+            unsupported_objects.append(k)
         else:
             to_serialize.update({k: object_json})
 
+    _logger.info(
+        "Serialization of the following objects are not supported and will not be logged with "
+        f"your model: {', '.join(unsupported_objects)}"
+    )
```

Comment:
```suggestion
    if unsupported_objects:
        _logger.info(
            "Serialization of the following objects are not supported and will not be logged with "
            f"your model: {', '.join(unsupported_objects)}"
        )
```

Can we only show the warning when there is unsupported object?